components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/sshuseraccesschecktrigger/sshuseraccesschecktrigger.go

```diff
@@ -0,0 +1,57 @@
+package sshuseraccesschecktrigger
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/configutils"
+	"github.com/chef/automate/lib/logger"
+)
+
+type SshUserAccessCheck struct {
+	host string
+	port string
+	log  logger.Logger
+}
+
+func NewSshUserAccessCheck(log logger.Logger, port string) *SshUserAccessCheck {
+	return &SshUserAccessCheck{
+		log:  log,
+		host: constants.LOCAL_HOST_URL,
+		port: port,
+	}
+}
+
+func (ss *SshUserAccessCheck) Run(config models.Config) []models.CheckTriggerResponse {
+
+	ss.log.Info("Performing SSH user access check from batch check ")
+	count := config.Hardware.AutomateNodeCount + config.Hardware.ChefInfraServerNodeCount +
```

Comment:
If there a same node is being used for different purpose like same node for both Automate node as well as Infra Node, the count calculated like this will be more than the number of unique nodes. This way you will be in trouble while looping though the count between lines 51 to 54. Ideally you should be appending channels as you make the go calls to make sure you looping the same number of channels as you have used while calling go routines.

Reply:
We discussed on the call regarding this.